frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -147,6 +183,44 @@ export const getEventSourceData = (source: string) => {
   return eventSourceData[source];
 };
 
+export const getEventSourceYaml = (namespace: string, eventSource: string): K8sResourceKind => {
+  let eventSourceModel: K8sKind;
+  const name = _.kebabCase(eventSource);
+  const defaultModel = {
+    apiGroup: KNATIVE_EVENT_SOURCE_APIGROUP_DEP,
+    apiVersion: 'v1alpha1',
+    kind: '',
+    label: '',
+    labelPlural: '',
+    plural: '',
+    abbr: '',
+  };
+  switch (eventSource.toLowerCase()) {
+    case 'camelsource':
+      eventSourceModel = EventSourceCamelModel;
+      break;
+    default:
+      eventSourceModel = defaultModel;
+  }
+  return {
+    apiVersion: `${eventSourceModel.apiGroup}/${eventSourceModel.apiVersion}`,
+    kind: eventSourceModel.kind,
+    metadata: {
+      name,
+      namespace,
+    },
+    spec: {
+      sink: {
+        ref: {
+          apiVersion: `${ServiceModel.apiGroup}/${ServiceModel.apiVersion}`,
+          kind: ServiceModel.kind,
+          name: '',
+        },
+      },
+    },
+  };
+};
```

Comment:
@nemesis09 why is this needed, can we make use of `yamlData` itself in initialValues

Reply:
updated code to use `getEventSourcesDepResource` for loading yaml initial values